### ISSUE TITLE
Suggestion on non-symmetric non-positive-definite real-valued generalized problems

### ISSUE THREAD

**Issue by dariomangoni** [opencollab/arpack-ng#313]
Hi guys,
I've been trying to solve the following problem:
![immagine](https://user-images.githubusercontent.com/8025529/122951902-eddd2d00-d37d-11eb-8662-6b50bb0d7ac7.png)
where
![immagine](https://user-images.githubusercontent.com/8025529/122951946-f5043b00-d37d-11eb-94a2-9d844034f0e5.png)

I anticipate that M_tilde and K_tilde matrices are real, but are NOT positive-definite and NOT symmetric, none of them.
Tuckling multibody+finite-elements...

I read from ARPACK guide that `dnaupd`+`dneupd`, that are the "most" fit for the purpose, require that my M_tilde should be symmetric and positive-definite
![immagine](https://user-images.githubusercontent.com/8025529/122952991-a99e5c80-d37e-11eb-9768-0c0714f2f8b6.png)

I tried nonetheless to run the problem with `dnaupd` realizing that returns ido=99 and info=0.
However, when calling `dneupd`, the `dr` vector returns just a long list of eigvalues all really close to `sigmar` and non-sense eigenvectors...

I know I'm in a quite desperate situation, but any suggestion on how to tackle such kind of problems is hugely appreciated.

Using ARPACK-NG 3.8.0

**Comment by caliarim** [comment]
Hi, are -M_bar and/or K_bar SPD?

**Comment by dariomangoni** [comment]
hi @caliarim, unfortunately no;
in aeroelasticity we cannot ensure that neither K_bar, M_bar or C_bar are symmetric nor that are positive-definite.
We are looking for solutions for the most general problems that we could encounter, but I'm struggling to find a proper method for handling such cases.

I think that using Krylov-Schur methods I might get a way out but, since we are already using ARPACK, I wanted to double check if we could handle these cases with IRAM so to avoid adding any additional dependency.

**Comment by caliarim** [comment]
If I am not wrong, inv(M_tilde)*K_tilde = [0,I;-inv(M_bar)*K_bar,-inv(M_bar)*C_bar]. So, you can turn it into a regular eigenvalue problem, provided that M_bar is not singular.

**Comment by dariomangoni** [comment]
Yes, you are right and we are already doing so, however this causes a strong filling up: the starting matrices are sparse, while the inversion make them quite dense. Is there any chance to leave them sparse?

**Comment by fghoussen** [comment]
Playing with workspace size / shift / invert and / or solvers (BiCG instead of LLt for instance possibly with PC) when mode > 1 may help : `arpackmm` maybe convienient for such testing.
https://github.com/opencollab/arpack-ng/blob/0a655d918d558ef97881db2760668b8d1e3b4bc1/EXAMPLES/MATRIX_MARKET/README#L18
Arpack is maintained but no more developed so if you don't find a solution here, you may give a try to slepc (LOPSI ? SRRIT ? QMRPACK ? not sure if these are neither stable, nor designed for sparse [or dense ?] matrices)
